File: snippets/javascript/thread-loop.js
// CFX Thread Loop Pattern (JavaScript)
// Use setTick for recurring logic. Always await Delay() to yield.
// NEVER use Delay(0) unless you need per-frame execution.

setTick(async () => {
    await Delay(1000); // Check every 1 second (adjust as needed)

    const playerPed = PlayerPedId();
    const playerCoords = GetEntityCoords(playerPed, false);

    // Your logic here
});

File: snippets/lua/thread-loop.lua
-- CFX Thread Loop Pattern
-- Use this for periodic checks. Adjust Wait() time based on your needs.
-- NEVER use Wait(0) unless you need per-frame execution (drawing, input).

CreateThread(function()
    while true do
        Wait(1000) -- Check every 1 second (adjust as needed)

        local playerPed = PlayerPedId()
        local playerCoords = GetEntityCoords(playerPed)

        -- Your logic here

    end
end)
